src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,506 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
 things. If there are no guarantees we can issue warnings instead of failures. Ideas:
   - No *more* than the provided maxAnisotropy samples are used, by testing how many unique
     sample values come out of the sample operation.
-  - Result with and without anisotropic filtering is different (if the hardware supports it).
   - Check anisotropy is done in the correct direciton (by having a 2D gradient and checking we get
     more of the color in the correct direction).
-More generally:
-  - Test very large and very small values (even if tests are very weak).
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
 import { GPUTest } from '../../../gpu_test.js';
+import { mipSize } from '../../../util/texture/subresource.js';
 
-export const g = makeTestGroup(GPUTest);
+const kRTSize = 16;
+const kBytesPerRow = 256;
+const xm = kRTSize / 2;
+const kColorAttachmentFormat = 'rgba8unorm';
+const kTextureFormat = 'rgba8unorm';
+const colors = [
+  new Uint8Array([0xff, 0x00, 0x00, 0xff]), // miplevel = 0
+  new Uint8Array([0x00, 0xff, 0x00, 0xff]), // miplevel = 1
+  new Uint8Array([0x00, 0x00, 0xff, 0xff]), // miplevel = 2
+];
+const checkerColors = [
+  new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+  new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+];
+
+// renders texture a slanted plane placed in a specific way
+class SamplerAnisotropicFilteringSlantedPlaneTest extends GPUTest {
+  private compareUint8ArrayResult(a: Uint8Array, b: Uint8Array) {
+    if (a.byteLength !== b.byteLength) {
+      return false;
+    }
+    for (let i = 0, len = a.length; i < len; i++) {
+      if (a[i] !== b[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  checkCheckerBoardResults(results: Uint8Array[]) {
+    this.eventualAsyncExpectation(async niceStack => {
```

Comment:
I think eventualAsyncExpectation isn't needed unless the inner function is actually async, so here you could just not use it and use regular `t.expect` and so on.